system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -4,22 +4,27 @@ const {FormattedMessage} = require("react-intl");
 
 const TopSitesPerfTimer = require("./TopSitesPerfTimer");
 const TopSitesEdit = require("./TopSitesEdit");
-const TopSite = require("./TopSite");
+const {TopSite, TopSitePlaceholder} = require("./TopSite");
 
-const TopSites = props => (<TopSitesPerfTimer>
-  <section className="top-sites">
-    <h3 className="section-title"><span className={`icon icon-small-spacer icon-topsites`} /><FormattedMessage id="header_top_sites" /></h3>
-    <ul className="top-sites-list">
-      {props.TopSites.rows.slice(0, props.TopSitesCount).map((link, index) => link && <TopSite
-        key={link.guid || link.url}
-        dispatch={props.dispatch}
-        link={link}
-        index={index}
-        intl={props.intl} />)}
-    </ul>
-    <TopSitesEdit {...props} />
-  </section>
-</TopSitesPerfTimer>);
+const TopSites = props => {
+  const realTopSites = props.TopSites.rows.slice(0, props.TopSitesCount);
+  const placeholderCount = props.TopSitesCount - realTopSites.length;
```

Comment:
Do we no longer allow "holes" in the top sites? Say the user pinned something in the very last slot (#12) but only has 6 other top sites. What does the UI show for that? Does it collapse it? I guess I can test it out.